File: src/nomanssave/cU.java
package nomanssave;

import java.io.File;
import javax.swing.Icon;
import javax.swing.filechooser.FileView;

class cU extends FileView {
   // $FF: synthetic field
   final cT gw;

   cU(cT var1) {
      this.gw = var1;
   }

   public Icon getIcon(File var1) {
      String var2 = var1.getName();
      return (Icon)(var2.endsWith(".sh0") ? cT.as() : super.getIcon(var1));
   }

   public String getName(File var1) {
      String var2 = var1.getName();
      return var2.endsWith(".sh0") ? var2.substring(0, var2.length() - 4) : var2;
   }
}

File: NMSSaveEditor/nomanssave/mixed/cU.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;

namespace NMSSaveEditor
{

#if PORT_COMPLETE


public class cU : FileView {
   public string Name => getName();
   public cT gw;

   public cU(cT var1) {
      this.gw = var1;
   }

   public Icon getIcon(FileInfo var1) {
      string var2 = var1.Name;
      return (Icon)(var2.EndsWith(".sh0") ? cT.@as() : base.getIcon(var1));
   }

   public string getName(FileInfo var1) {
      string var2 = var1.Name;
      return var2.EndsWith(".sh0") ? var2.Substring(0, var2.length() - 4) : var2;
   }
}


#else

public class cU
{
   public cU() { }
   public cU(params object[] args) { }
   public string Name = "";
   public cT gw = default;
   public Icon getIcon(FileInfo var1) { return default; }
   public string getName(FileInfo var1) { return ""; }
}

#endif

}
